TestResultSummaryService/Database.js

```diff
@@ -159,27 +165,61 @@ class Database {
         if (query.platform)
             buildNameRegex = `${buildNameRegex}${query.platform}`;
 
-        const result = await this.aggregate([
-            { $match: matchQuery },
-            {
-                $graphLookup: {
-                    from: 'testResults',
-                    startWith: '$_id',
-                    connectFromField: '_id',
-                    connectToField: 'parentId',
-                    as: 'childBuilds',
+        // Call the routing function to get the specific aggregation
+        const specificAggregation = this.getSpecificAggregation(type, buildNameRegex, query);
+
+        try {
+            const result = await this.aggregate([
+                { $match: matchQuery },
+                {
+                    $graphLookup: {
+                        from: 'testResults',
+                        startWith: '$_id',
+                        connectFromField: '_id',
+                        connectToField: 'parentId',
+                        as: 'childBuilds',
+                    },
                 },
-            },
-            {
-                $project: {
-                    childBuilds: '$childBuilds',
+                {
+                    $project: {
+                        childBuilds: '$childBuilds',
+                    },
                 },
-            },
+                ...specificAggregation,
+            ]);
+            return result[0] || {};
+        } catch (error) {
+            console.error('Error:', error);
+        }
+    }
+
+    /**
+     * Routing function to get the specific aggregation based on the type.
+     * @param {string} type - The type of request (e.g., 'totals', 'rerunDetails', 'jobsDetails').
+     * @param {string} buildNameRegex - The build name regex.
+     * @param {Object} query - The query object.
+     * @returns {Array} - The specific aggregation steps.
+     */
+    getSpecificAggregation(type, buildNameRegex, query) {
+        switch (type) {
+            case 'totals':
+                return this.getTotalsAggregation(query, buildNameRegex);
+            case 'rerunDetails':
+                return this.getRerunDetailsAggregation();
+            case 'jobsDetails':
+                return this.getJobsDetailsAggregation();
+            default:
+                throw new Error(`Unknown type: ${type}`);
+        }
+    }
```

Comment:
getSpecificAggregation() calls different functions based on the switch type. Each function has its own mandate and purpose. I do not think we should create getSpecificAggregation() and use the switch case here.

If you really want to utilize the common code in testResultsBaseAggregation(), pass specificAggregation in testResultsBaseAggregation().